apps/web/src/common/key-map.ts

```diff
@@ -98,7 +98,11 @@ const KEYMAP = [
   {
     keys: ["ctrl+w", "command+w"],
     description: "Close active tab",
-    action: () => useEditorStore.getState().closeActiveTab()
+    action: () => {
+      const activeTab = useEditorStore.getState().getActiveTab();
+      if (activeTab?.pinned) return;
```

Comment:
Let's follow the behavior of other apps (e.g. VSCode, Zed). Pressing Ctrl + W on a pinned tab activates the previously activated tab instead of doing nothing.